Lib/typing.py

```diff
@@ -999,16 +999,37 @@ def __dir__(self):
 
 
 class _GenericAlias(_BaseGenericAlias, _root=True):
-    def __init__(self, origin, params, *, inst=True, name=None,
+    """The type of parameterized generics.
+
+    That is, for example, `type(List[int])` is `_GenericAlias`.
+
+    Objects which are instances of this class include:
+    * Parameterized container types, e.g. `Tuple[int]`, `List[int]`
+      * Note that native container types, e.g. `tuple`, `list`, use
+        `types.GenericAlias` instead.
+    * Parameterized classes:
+        T = TypeVar('T')
+        class C(Generic[T]): pass
+        # C[int] is a _GenericAlias
+    * `Callable` aliases, generic `Callable` aliases, and
+      parameterized `Callable` aliases:
+       T = TypeVar('T')
+       # _CallableGenericAlias inherits from _GenericAlias.
+       A = Callable[[], None]  # _CallableGenericAlias
+       B = Callable[[T], None]  # _CallableGenericAlias
+       C = B[int]  # _CallableGenericAlias
```

Comment:
And also instances of various special forms: Final[int], ClassVar[int], TypeGuard[int].

Reply:
Ah, nice - thanks.